src/Grpc.ChannelConfiguration/GrpcServiceMethodConfigurationExtensions.cs

```diff
@@ -0,0 +1,281 @@
+    }
+
+    /// <summary>
+    /// Customizes the <see cref="IHttpClientBuilder" /> to configure all service methods according to the values in <paramref name="section"/>.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHttpClientBuilder" /> to configure.</param>
+    /// <param name="section">The instance of <see cref="IConfigurationSection" /> containing the configuration to attach to all methods.</param>
+    /// <returns>The configured <see cref="IHttpClientBuilder" />.</returns>
+    /// <exception cref="ArgumentException">Thrown when the method is configured with both a <see cref="RetryPolicy" /> and a <see creft="HedgingPolicy" />.</exception>
+    public static IHttpClientBuilder ConfigureDefaultServiceMethod(this IHttpClientBuilder builder, IConfigurationSection section)
+    {
+        _ = builder ?? throw new ArgumentNullException(nameof(builder));
+
+        return ConfigureServiceMethod(builder, MethodName.Default, section);
+    }
+
+    /// <summary>
+    /// Customizes the <see cref="IHttpClientBuilder" /> to configure all service methods with the specified <see cref="RetryPolicy" />.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHttpClientBuilder" /> to configure.</param>
+    /// <param name="retryPolicy">The <see cref="RetryPolicy" /> to attach to all service methods.</param>
+    /// <returns>The configured <see cref="IHttpClientBuilder" />.</returns>
+    public static IHttpClientBuilder ConfigureDefaultServiceMethod(this IHttpClientBuilder builder, RetryPolicy retryPolicy)
+    {
+        _ = builder ?? throw new ArgumentNullException(nameof(builder));
+
+        _ = retryPolicy ?? throw new ArgumentNullException(nameof(retryPolicy));
+
+        return ConfigureServiceMethod(builder, MethodName.Default, retryPolicy);
+    }
+
+    /// <summary>
+    /// Customizes the <see cref="IHttpClientBuilder" /> to configure all service methods with the specified <see cref="HedgingPolicy" />.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHttpClientBuilder" /> to configure.</param>
+    /// <param name="hedgingPolicy">The <see cref="HedgingPolicy" /> to attach to all service methods.</param>
+    /// <returns>The configured <see cref="IHttpClientBuilder" />.</returns>
+    public static IHttpClientBuilder ConfigureDefaultServiceMethod(this IHttpClientBuilder builder, HedgingPolicy hedgingPolicy)
+    {
+        _ = builder ?? throw new ArgumentNullException(nameof(builder));
+
+        _ = hedgingPolicy ?? throw new ArgumentNullException(nameof(hedgingPolicy));
+
+        return ConfigureServiceMethod(builder, MethodName.Default, hedgingPolicy);
+    }
+
+    private static T LoadPolicy<T>(IConfigurationSection section, string sectionName)
+    {
+        _ = section ?? throw new ArgumentNullException(nameof(section));
+
+        return section.GetSection(sectionName).Get<T>();
+    }
+
+    private static MethodName ParseKey(string key)
```

Comment:
Is this just a slightly more complicated way to do a String.Split just so you can use the range operator? 😄

Reply:
Rewritten in the boring way.